### PR TITLE
Add fu_common_sum8() common functionality

### PR DESCRIPTION
**Pull request by hughsie**
We now have 9 different plugins all using this functionality, and we're
about to add one more. Move this into common code so that all the
plugins are using the same endian and bufsz-safe versions.

Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [X] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

File: plugins/ccgx/fu-ccgx-firmware.c [hunk 1]
```diff
@@ -238,7 +227,7 @@ fu_ccgx_firmware_parse_md_block(FuCcgxFirmware *self, FwupdInstallFlags flags, G
 	}
 	for (guint i = 0; i < self->records->len - 1; i++) {
 		rcd = g_ptr_array_index(self->records, i);
-		checksum_calc += fu_ccgx_firmware_record_calc_checksum(rcd);
+		checksum_calc += fu_common_sum16_bytes(rcd->data);
```
Comment (d4s) [inline plugins/ccgx/fu-ccgx-firmware.c:230]: `fu_ccgx_firmware_record_calc_checksum()` returns **guint8**

File: plugins/wacom-usb/fu-wac-device.c [hunk 2]
```diff
@@ -583,7 +583,7 @@ fu_wac_device_write_firmware(FuDevice *device,
 		}
 
 		/* calculate expected checksum and save to device RAM */
-		csum_local[i] = fu_wac_calculate_checksum32le_bytes(blob_block);
+		csum_local[i] = fu_common_sum32w_bytes(blob_block, G_LITTLE_ENDIAN);
```
Comment (d4s) [inline plugins/wacom-usb/fu-wac-device.c:586]: `fu_wac_calculate_checksum32le_bytes()` returns guint32 as G_LITTLE_ENDIAN while new API is a host-based.

**Review by d4s — CHANGES_REQUESTED** [review]

**Comment by d4s** [comment]
Tested with SteelSeries device only -- works well.

File: plugins/ccgx/fu-ccgx-firmware.c (hunk 1 above)
Comment (hughsie) [reply plugins/ccgx/fu-ccgx-firmware.c:230]: Fixed, thanks; Good catch!

File: plugins/wacom-usb/fu-wac-device.c (hunk 2 above)
Comment (hughsie) [reply plugins/wacom-usb/fu-wac-device.c:586]: Yes, agreed!

    - 		csum_local[i] = fu_common_sum32w_bytes(blob_block, G_LITTLE_ENDIAN); 
    +		csum_local[i] = GUINT32_TO_LE(fu_common_sum32w_bytes(blob_block, G_LITTLE_ENDIAN));

right?

**Review by d4s — APPROVED** [review]

**Pull request merged**